### ISSUE TITLE
cannot run train_detection.sh

### ISSUE THREAD

**Issue by LeonJWH** [zhaoweicai/cascade-rcnn#5]
When I try to run you training script to train a vgg cascade faster rcnn, I encounter a error like this:

> I0426 15:14:21.523888 14111 layer_factory.hpp:77] Creating layer data
> F0426 15:14:21.523939 14111 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: DetectionData (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, BboxAccuracy, Bias, BoxGroupOutput, Concat, ContrastiveLoss, Convolution, Crop, Data, DecodeBBox, Deconvolution, DetectionEvaluate, DetectionGroupAccuracy, DetectionGroupLoss, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, FrcnnOutput, HDF5Data, HDF5Output, HardMining, HingeLoss, Im2col, InfogainLoss, InnerProduct, Input, LRN, LSTM, LSTMUnit, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, PSROIPooling, Parameter, Pooling, Power, ProposalTarget, Python, RNN, ROIAlign, ROIPooling, ReLU, Reduction, Reshape, SPP, Scale, ScaleRoute, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, SmoothL1Loss, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile)
> *** Check failure stack trace: ***
>     @     0x7f87d5321daa  (unknown)
>     @     0x7f87d5321ce4  (unknown)
>     @     0x7f87d53216e6  (unknown)
>     @     0x7f87d5324687  (unknown)
>     @     0x7f87d59a7f89  caffe::LayerRegistry<>::CreateLayer()
>     @     0x7f87d59aa488  caffe::Net<>::Init()
>     @     0x7f87d59ac262  caffe::Net<>::Net()
>     @     0x7f87d5b3eab0  caffe::Solver<>::InitTrainNet()
>     @     0x7f87d5b3efc3  caffe::Solver<>::Init()
>     @     0x7f87d5b3f27f  caffe::Solver<>::Solver()
>     @     0x7f87d5b31241  caffe::Creator_SGDSolver<>()
>     @           0x414437  caffe::SolverRegistry<>::CreateSolver()
>     @           0x40dbd5  train()
>     @           0x40a02c  main
>     @     0x7f87d3b81f45  (unknown)
>     @           0x40aa6b  (unknown)
>     @              (nil)  (unknown)
> Aborted (core dumped)

I have checked the **caffe.proto** and the parameters of _detection_data_layer_ are already registered,  can you give me some advise?

**Comment by zhaoweicai** [comment]
The problem may be relevant with OpenCV. This whole data layer depends on OpenCV. If you set USE_OPENCV := 0 in the Makefile.config and it will not be complied. Try to turn on the OpenCV.

**Comment by LeonJWH** [comment]
I have set `USE_OPENCV = 1`, in the **Makefile.config**, and the compilation goes well without warnings and errors.
![2018-04-27 10 08 11 2](https://user-images.githubusercontent.com/17867054/39340838-f4256e3c-4a02-11e8-814a-e8bd69a50060.png)

**Comment by zhaoweicai** [comment]
Hi @LeonJWH, I tried you Makefile.config, but I can't reproduce your error. Could you check your compilation log if the DetectionData layer was complied?

**Comment by LeonJWH** [comment]
thank you！
In the compilation log, the DetectionData layer is compiled, but the problem still exits. I also tried to run your repo on another machine, it went well. So I guess is my machine's problem, sorry to bother you.
